iib/web/models.py

```diff
@@ -852,16 +859,10 @@ def from_json(cls, kwargs, batch=None):
                 '"bundles" should be either an empty array or an array of non-empty strings'
             )
 
-        # Check if no bundles `from_index and `binary_image` are specified
-        # if no bundles and and no from index then a empty index will be created
-        # if no binary image and just a from_index then we are not updating anything and it would
-        # be a no-op
-        if not request_kwargs.get('bundles') and (
-            not request_kwargs.get('from_index') or not request_kwargs.get('binary_image')
-        ):
-            raise ValidationError(
-                '"from_index" and "binary_image" must be specified if no bundles are specified'
-            )
+        # Check if no bundles and `from_index is specified
+        # if no bundles and and no from index then a empty index will be created which is a no-op
+        if not request_kwargs.get('bundles') and not request_kwargs.get('from_index'):
+            raise ValidationError('"from_index" must be specified if no bundles are specified')
```

Comment:
This seems like a change in behavior. Don't we still want to allow creating empty index images?

Reply:
We were not allowing to create an empty index image in the previous behavior too, hence this change to keep that behavior same. We do not want a request with no `bundles` and no `from_index`. Wasn't that the case previously?

Reply:
You are right. I misread the code.